include/seastar/util/process.hh

```diff
@@ -0,0 +1,112 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Copyright (C) 2022 Kefu Chai ( tchaikov@gmail.com )
+ */
+
+#pragma once
+
+#include <sys/types.h>
+#include <algorithm>
+#include <filesystem>
+#include <initializer_list>
+#include <iterator>
+#include <string_view>
+#include <utility>
+#include <variant>
+#include <vector>
+#include <fmt/format.h>
+#include <seastar/core/iostream.hh>
+#include <seastar/core/sstring.hh>
+
+namespace seastar::experimental {
+
+/// The optional parameters for spawning a subprocess
+///
+/// \note see \c execve(2) for more details on \c argv and \c env.
+struct spawn_parameters {
+    /// The arguments passed to the program
+    std::vector<sstring> argv;
+    /// The environment variables for the program
+    std::vector<sstring> env;
+    spawn_parameters(std::initializer_list<std::string_view> argv)
+        : argv(argv.begin(), argv.end()) {}
+    spawn_parameters(std::initializer_list<std::string_view> argv,
+                     std::initializer_list<std::pair<std::string_view, std::string_view>> envs)
+        : argv(argv.begin(), argv.end())
+    {
+        std::transform(envs.begin(), envs.end(), std::inserter(env, env.end()),
+                       [](const auto& e) {
+                           return fmt::format("{}={}", e.first, e.second);
+                       });
+    }
```

Comment:
And things like that are best left to the user, or maybe a helper. We should be providing a thin layer and adapt to the async nature of Seastar, not try to change the posix API.

Reply:
dropped. thought it was also important to have a user friendly API, but yeah, it might diverge from the goal of Seastar, and could open a door of more shim-like APIs.